nion/eels_analysis/BackgroundModel.py

```diff
@@ -119,8 +119,41 @@ def __unused_perform_fit(self, xs: numpy.ndarray, ys: numpy.ndarray, fs: numpy.n
         series = typing.cast(typing.Any, numpy.polynomial.polynomial.Polynomial.fit(xs, transform_data(ys), self.deg))
         return untransform_data(series(fs))
 
+class TwoAreaBackgroundModel(AbstractBackgroundModel):
+
+    def __init__(self, background_model_id: str, params_func = None, model_func = None, title: str = None):
+        super().__init__(background_model_id, title)
+        self.model_func = model_func
+        self.params_func = params_func
+
+
+    def _perform_fits(self, xs: numpy.ndarray, yss: numpy.ndarray, fs: numpy.ndarray) -> numpy.ndarray:
+        if not len(xs) % 2 == 0:
+            yss = yss[...,:-1]
+            xs = xs[:-1]
+        half_interval = len(xs) // 2
+        areas_1 = numpy.trapz(yss[...,:half_interval], xs[:half_interval], axis=1)
+        areas_2 = numpy.trapz(yss[...,half_interval:], xs[half_interval:], axis=1)
```

Comment:
I think a potentially better way to pull data out is to use [...,-half_interval:]. I think that avoids the issue with odd length arrays on lines 131-133 -- although you're essentially skipping the middle pixel rather than the end pixel in that case.

Reply:
Hmmm. I'm a little concerned about sometimes skipping the middle pixel and sometimes not, because the algorithm also uses the "center" x value. For fit regions with small numbers of pixels, the even and odd cases could give different results. I'm not sure how much of an issue it is to trim the end pixel in the odd case. I'm guessing that the vast majority of cases, the user will select a large enough interval that it will be inconsequential.

Reply:
As an alternative: [...,half_interval:2*half_interval]. This also avoids the trim at the beginning.